subprojects/test-runner/test-report-artifacts/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt

```diff
@@ -34,20 +34,24 @@ public object TestArtifactsProviderFactory {
     private fun create(provider: ReportDirProvider) =
         DirectTestArtifactsProvider(provider)
 
-    // android API's are unavailable here
+    // Android API's are unavailable here
     @Suppress("SdCardPath")
     public fun createForAdbAccess(
+        api: Int,
         appUnderTestPackage: String,
         name: TestName,
     ): TestArtifactsProvider {
-        val dataPath = "/sdcard/Android/data/$appUnderTestPackage/files"
+        val dataPath = if (api >= 30) {
+            "/storage/emulated/0/Android/media/$appUnderTestPackage"
+        } else {
+            "/sdcard/Android/data/$appUnderTestPackage/files"
+        }
```

Comment:
Need advice on hiding these implementation details.
This knowledge is on different layers.
Glad to discuss it on call.

Reply:
yes, let's discuss it after tomorrow's call

Reply:
https://github.com/avito-tech/avito-android/pull/1156